### ISSUE TITLE
[BUG] Model.create and Model.get differ on custom types

### ISSUE THREAD

**Issue by pmezard** [dynamoose/dynamoose#1211]
### Summary:

If you have a schema containing an attribute like:
```
someDate: {
  type: Date
}
```
and create a document using the derived model like:
```
const m = someModel.create({
   ...
   someDate: Date.now(),
})
```
Then `m.someDate` is a `number`.

If later you get the document with:
```
const m2 = someModel.get(mId)
```
Then `m2.someDate` is a `Date`.

According to this Slack thread https://dynamoose.slack.com/archives/CG4B7RL8N/p1621584884007300 , it should always be a `Date`.

I think the issue is `conformToSchema` is called in the `get` code path, but not in the `create` one.

### Environment:

Operating System: MacOS
Operating System Version:  Big Sur 11.2.3
Node.js version (`node -v`): v15.14.0
NPM version: (`npm -v`): 6.14.13
Dynamoose version: 2.7.3

### Other:
- [ x] I have read through the Dynamoose documentation before posting this issue
- [ x] I have searched through the GitHub issues (including closed issues) and pull requests to ensure this issue has not already been raised before
- [ x] I have searched the internet and Stack Overflow to ensure this issue hasn't been raised or answered before
- [ x] I have tested the code provided and am confident it doesn't work as intended
- [ x] I have filled out all fields above
- [ x] I am running the latest version of Dynamoose

Thanks !

**Comment by Drarig29** [comment]
You are correct in that a `conformToSchema` call is missing, and it works if I do this:

```js
const fixed = await item.conformToSchema({
    type: 'fromDynamo',
    customTypesDynamo: true,
});
```

**Comment by thinkjones** [comment]
I am also experiencing the same problem. When you call `model.create` you get a different object then when you call `model.get` for the same id. Happens when you are using `get` and `set` methods to transform data.

```
  pk: {
    type: String,
    hashKey: true,
    map: 'id',
    required: true,
    set: (value) => {
      console.log('ID pk set', value)
      return `ID#${value}`
    },
    get: (value) => {
      console.log('ID pk get', value)
      return extractIdFromKey(value.toString())
    }
  },
  ```

**Comment by jimmyn** [comment]
I am experiencing the same issue.
